lib/hyperion_test/fake_server/config.rb

```diff
@@ -0,0 +1,36 @@
+class Hyperion
+  class FakeServer
+    class Config
+      # this is passed to the block to allow the caller to configure the fake server
+
+      include Hyperion::Headers
+      include Hyperion::Logger
+
+      def rules
+        @rules ||= []
+      end
+
+      # allow(route)
+      # allow(method, path, headers={})
+      def allow(*args, &handler)
+        rule = allowed_rule(args, handler)
+        rules << rule
+        log_stub(rule)
+      end
+
+      private
+
+      def allowed_rule(args, handler)
+        if args.size == 1 && args.first.is_a?(RestRoute)
+          route = args.first
+          Rule.new(MimicRoute.new(route.method, route.uri.path), route_headers(route), handler, route)
+        else
+          # TODO: deprecate this
+          method, path, headers = args
+          headers ||= {}
+          Rule.new(MimicRoute.new(method, path), headers, handler, nil)
+        end
+      end
+    end
+  end
+end
```

Comment:
That was our original plan but it fell by the wayside. The problem is that you need to keep the two gems in sync, which reduces our agility. As a compromise, the test-related code still lives in the hyperion gem, but it is only loaded when you `require 'hyperion_test'`. Assuming no production code `require`s `hyperion_test`, none of the test-related dependencies will be loaded in production, although the gem dependencies will be part of bundle.

I'll make a note of this in the README